core/src/main/java/org/apache/druid/guice/GuiceAnnotationIntrospector.java

```diff
@@ -101,13 +103,28 @@ public Object findInjectableValueId(AnnotatedMember m)
     // delegate creators. I'm not 100% sure why it's not called, but guess it's because the argument
     // is some Java type that Jackson already knows how to deserialize. Since there is only one argument,
     // Jackson perhaps is able to just deserialize it without introspection.
+
     if (ac instanceof AnnotatedParameter) {
       final AnnotatedParameter ap = (AnnotatedParameter) ac;
       if (ap.hasAnnotation(JsonProperty.class)) {
         return JsonIgnoreProperties.Value.empty();
       }
     }
 
-    return JsonIgnoreProperties.Value.forIgnoredProperties("");
+    // A map can have empty string keys e.g. https://github.com/apache/druid/issues/10859. By returning empty ignored
+    // list for map, we can allow for empty string keys in a map. A nested class within map
+    // can still be annotated with JacksonInject and still be non-deserializable from user input
+    // Refer to {@link com.fasterxml.jackson.databind.deser.BasicDeserializerFactory.createMapDeserializer} for details
+    // on how the ignored list is passed to map deserializer
+    if (ac instanceof AnnotatedClass) {
```

Comment:
I'm not sure whether this if-else is necessary. Actually, the bug reported by https://github.com/apache/druid/issues/10859 is caused by ignoring empty string keys in a map when deserialize org.apache.druid.query.extraction.MapLookupExtractor instead of Map

Reply:
this block will ensure that we don't ignore string keys if the class is a map type.

Reply:
if we have a Map type field inside a class (just like Map is a field inside org.apache.druid.query.extraction.MapLookupExtractor), the bug reported by #10859 means when we deserialize org.apache.druid.query.extraction.MapLookupExtractor, the empty string key in Map will be ignored. So in this case, AnnotatedClass here is MapLookupExtractor instead Map. 

To verify it, I remove this block and it works well.

Reply:
@Gabriel39 the bug is that the empty string key in a `Map` is not deserialized. When you have a class that has a map as a field, this method will be called for both the outer class and the map. `MapLookupExtractionFnSerDeTest` is improved in this PR to test the case where a `MapLookupExtractor` has a map that has an empty string key. That test will fail without this block.